pkg/pillar/types/errortime.go

```diff
@@ -12,19 +12,93 @@ import (
 
 // Common error with timestamp
 
+// ErrorSeverity tells the severity type, must be in sync with info api
+type ErrorSeverity int32
+
+const (
+	// ErrorSeverityUnspecified severity unspecified
+	ErrorSeverityUnspecified ErrorSeverity = 0
+	// ErrorSeverityNotice severity notice
+	ErrorSeverityNotice ErrorSeverity = 1
+	// ErrorSeverityWarning severity warning
+	ErrorSeverityWarning ErrorSeverity = 2
+	// ErrorSeverityError severity error
+	ErrorSeverityError ErrorSeverity = 3
+)
+
+//GetErrorSeverity returns ErrorSeverity based on retry count and time spend
+func GetErrorSeverity(retryCount int, timeSpend time.Duration) ErrorSeverity {
+	if retryCount > 20 || timeSpend > 2*time.Hour {
+		return ErrorSeverityError
+	}
+	if retryCount > 10 || timeSpend > time.Hour {
+		return ErrorSeverityWarning
+	}
+	return ErrorSeverityNotice
+}
+
+// ErrorEntityType contains the entity type, must be in sync with info api
+type ErrorEntityType int32
+
+const (
+	// ErrorEntityUnspecified Entity
+	ErrorEntityUnspecified ErrorEntityType = 0
+	// ErrorEntityBaseOs entity
+	ErrorEntityBaseOs ErrorEntityType = 1
+	// ErrorEntitySystemAdapter Entity
+	ErrorEntitySystemAdapter ErrorEntityType = 2
+	// ErrorEntityVault Entity
+	ErrorEntityVault ErrorEntityType = 3
+	// ErrorEntityAttestation Entity
+	ErrorEntityAttestation ErrorEntityType = 4
+	// ErrorEntityAppInstance Entity
+	ErrorEntityAppInstance ErrorEntityType = 5
+	// ErrorEntityPort Entity
+	ErrorEntityPort ErrorEntityType = 6
+	// ErrorEntityNetwork Entity
+	ErrorEntityNetwork ErrorEntityType = 7
+	// ErrorEntityNetworkInstance Entity
+	ErrorEntityNetworkInstance ErrorEntityType = 8
+	// ErrorEntityContentTree Entity
+	ErrorEntityContentTree ErrorEntityType = 9
+	// ErrorEntityContentBlob Entity
+	ErrorEntityContentBlob ErrorEntityType = 10
+	// ErrorEntityVolume Entity
+	ErrorEntityVolume ErrorEntityType = 11
+)
+
+// ErrorEntity contains the device entity details
+type ErrorEntity struct {
+	EntityType ErrorEntityType // entity type
+	EntityID   string          // entity uuid
+}
+
+// ErrorDescription contains error description to be passed into functions
+type ErrorDescription struct {
+	Error               string
+	ErrorSeverity       ErrorSeverity
+	ErrorRetryCondition string
+	ErrorEntities       []*ErrorEntity
+}
+
+// Copy creates new ErrorDescription as a copy of provided
+func (ed ErrorDescription) Copy() ErrorDescription {
+	c := ErrorDescription{Error: ed.Error, ErrorSeverity: ed.ErrorSeverity, ErrorRetryCondition: ed.ErrorRetryCondition}
+	for _, el := range ed.ErrorEntities {
+		c.ErrorEntities = append(c.ErrorEntities, &ErrorEntity{EntityID: el.EntityID, EntityType: el.EntityType})
+	}
+	return c
+}
+
 // ErrorAndTime is used by many EVE agents
 type ErrorAndTime struct {
-	Error     string
 	ErrorTime time.Time
```

Comment:
I think we can move ErrorTime into the ErrorDescription as well.

Reply:
Moved